src/bidi.jl

```diff
@@ -0,0 +1,23 @@
+import Base: size, getindex, setindex!
+
+#z indices are flipped for even-numbered timepoints
+struct BidiImageArray{T} <: AbstractArray{T,4}
+    A::AbstractArray{T,4}
+    z_size::Int
+end
+
+BidiImageArray(A::AbstractArray{T,4}) where {T} = BidiImageArray(A, size(A,3))
+
+size(B::BidiImageArray) = size(B.A)
+
+Base.IndexStyle(::Type{<:BidiImageArray}) = IndexCartesian()
+
+function getindex(B::BidiImageArray, I::Vararg{Int, 4})
+    t_ind = I[4]
+    ifelse(isodd(t_ind), getindex(B.A, I...), getindex(B.A, I[1], I[2], B.z_size - I[3] + 1, t_ind))
+end
+
+function setindex!(B::BidiImageArray, v, I::Vararg{Int, 4})
+    t_ind = I[4]
+    ifelse(isodd(t_ind), setindex!(B.A, v, I...), setindex!(B.A, v, I[1], I[2], B.z_size - I[3] + 1, I[4]))
```

Comment:
Likewise but worse. This will change *both* locations.

Reply:
Doh!  I really did understand `select` instead of branching, honest... Interesting that my tests didn't catch the double setting.